src/components/Grid.tsx

```diff
@@ -293,12 +292,34 @@ export default class Grid extends Component<GridProps, GridState> {
         }
     }
 
-    addEventListener(param1: any, param2: any, param3: any) {
-        this.eventListeners.push([param1, param2, param3])
+    addEventListener(param1: any, param2: any, param3: any, listenerId: string) {
+        this.eventListeners.push([param1, param2, param3, listenerId])
+        this.removeDuplicatedEventListeners(this.eventListeners)
+    }
+
+    removeEventListener(type: string, listenerId: string) {
+        this.eventListeners = this.eventListeners.filter((param: any[]) => {
+            if (param[0] !== type) {
+                return param
+            } else if (param[0] === type && type !== "componentDidUpdate" && param[3] !== listenerId) {
+                return param
+            } else if (param[0] === type && param[0] === "componentDidUpdate" && param[2] !== listenerId) {
+                return param
+            }
+        })
     }
 
-    removeEventListener(type: string, listener: any) {
-        this.eventListeners = this.eventListeners.filter((param: any[]) => param[0] !== type && param[2].toString() !== listener.toString())
+    removeDuplicatedEventListeners(eventListeners: any) {
```

Comment:
In my opinion the array of event listeners shouldn't be passed by param. Just iterate through `this.eventListeners`.